clash_lib/src/app/dispatcher/tracked.rs

```diff
@@ -515,18 +516,18 @@ impl Stream for TrackedDatagram {
 }
 
 impl Sink<UdpPacket> for TrackedDatagram {
-    type Error = std::io::Error;
+    type Error = watfaq_error::Error;
 
     fn poll_ready(
         mut self: Pin<&mut Self>,
         cx: &mut std::task::Context<'_>,
     ) -> Poll<Result<(), Self::Error>> {
         match self.close_notify.try_recv() {
-            Ok(_) => return Poll::Ready(Err(std::io::ErrorKind::BrokenPipe.into())),
+            Ok(_) => return Poll::Ready(Err(anyhow!("shutdown"))),
```

Comment:
these are actually IO related errors. i think it's fair to keep the io:Error type

Reply:
I am planning to remove Stream<Item = UdpPacket> + Sink things in code. I dont remember why i change Error type. Probably because io::Error dont cover all error types to the Structs which impl Stream<Item = UdpPacket> + Sink